Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,243 @@
+            return holderCount == 1 ? ($"{highestLastBalanceHolder} is the richest person. ¤{stringHighesLasttBalance}.") : ($"{highestLastBalanceHolder} are the richest people. ¤{stringHighesLasttBalance}.");
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            {
+                if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+                {
+                    return "N/A.";
+                }
+                int holderCount = 0;
+                float overallLowestLastBalance = float.MaxValue;
+                string currentBalanceHolder;
+                string overallLowestLastBalanceHolder = "";
+                bool isNegativeBalance = false;
+
+                foreach (string entry in peopleAndBalances)
+                {
+                    string[] userHistory = entry.Split(',');
+
+                    currentBalanceHolder = userHistory[0];
+
+                    float currentBalanceHolderFinalBalance = float.Parse(userHistory[^1]);
+
+                    if (overallLowestLastBalance > currentBalanceHolderFinalBalance)
+                    {
+                        overallLowestLastBalanceHolder = currentBalanceHolder;
+                        holderCount = 1;
+                    }
+
+                    if (overallLowestLastBalance == currentBalanceHolderFinalBalance && overallLowestLastBalanceHolder != currentBalanceHolder)
+                    {
+                        switch (holderCount)
+                        {
+                            case 1:
+                                {
+                                    overallLowestLastBalanceHolder += ($", {currentBalanceHolder}");
+                                    holderCount++;
+                                    break;
+                                }
+                            case 2:
+                                {
+                                    overallLowestLastBalanceHolder += ($" and {currentBalanceHolder}");
+                                    break;
+                                }
+                        }
+                    }
+                    overallLowestLastBalance = overallLowestLastBalance > currentBalanceHolderFinalBalance ? currentBalanceHolderFinalBalance : overallLowestLastBalance;
+
+                    isNegativeBalance = (overallLowestLastBalance < 0);
+                    
+                }
+                string stringHighesLasttBalance = Math.Abs(overallLowestLastBalance).ToString();
+                if (isNegativeBalance)
+                {
+                    return holderCount == 1 ? ($"{overallLowestLastBalanceHolder} has the least money. -¤{stringHighesLasttBalance}.") : ($"{overallLowestLastBalanceHolder} have the least money. -¤{stringHighesLasttBalance}.");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Split this 227 characters long line (which is greater than 200 authorized).](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4953253)